provider/snowflake.go

```diff
@@ -123,138 +125,110 @@ func (sf *snowflakeOfflineStore) UpdateTransformation(config TransformationConfi
-	logger.Info("Snowflake offline store creating resource table...")
-	if id.Type == Feature {
-		logger.Debugw("Snowflake Offline Store does not support creating resource tables for features")
-		return nil, nil
+	if len(opts) > 0 {
+		return nil, fferr.NewInvalidArgumentErrorf("snowflake offline store does not currently support resource options")
 	}
-	if err := id.check(Label); err != nil {
-		logger.Errorw("Failed to validate resource ID", "error", err)
+	containsCols, err := sf.checkSourceContainsResourceColumns(id, schema, opts...)
+	if err != nil {
 		return nil, err
 	}
-	if len(opts) == 0 || opts[0].Type() != SnowflakeDynamicTableResource {
-		logger.Errorw("Snowflake Offline Store requires Dynamic Table Resource Options")
-		return nil, fferr.NewInternalErrorf("Snowflake Offline Store requires Dynamic Table Resource Options")
-	}
-	opt, isSnowflakeConfigOpt := opts[0].(*ResourceSnowflakeConfigOption)
-	if !isSnowflakeConfigOpt {
-		logger.Errorw("Snowflake Offline Store requires Resource Config Options; received %T", opts[0])
-		return nil, fferr.NewInternalErrorf("Snowflake Offline Store requires Resource Config Options; received %T", opts[0])
-	}
-	tableConfig := opt.Config
-	if tableConfig == nil {
-		logger.Debugw("Resource table config for label resource is empty")
-		tableConfig = &metadata.SnowflakeDynamicTableConfig{}
-	}
-	logger.Debugw("Resource table config for label resource before Merge with Snowflake Config", "config", tableConfig)
-	resConfig := &metadata.ResourceSnowflakeConfig{
-		DynamicTableConfig: tableConfig,
-		Warehouse:          opt.Warehouse,
-	}
-	var snowflakeConfig pc.SnowflakeConfig
-	if err := snowflakeConfig.Deserialize(sf.sqlOfflineStore.Config()); err != nil {
-		logger.Errorw("Failed to deserialize snowflake config", "error", err)
-		return nil, err
+	if !containsCols {
+		return nil, fferr.NewInvalidArgumentErrorf("source table does not contain expected columns")
 	}
-	logger.Debugw("Snowflake catalog config for label resource", "config", snowflakeConfig.Catalog)
-	if err := resConfig.Merge(&snowflakeConfig); err != nil {
-		logger.Errorw("Failed to merge dynamic table config", "error", err)
-		return nil, err
+	return nil, nil
+}
+
+// checkSourceContainsResourceColumns checks that the source table contains the expected columns for the resource type
+// NOTE: due to the fact that we don't change the case of the columns inputted by the user, which is not an issue in the context
+// of SQL queries due to the face they're generally case insensitive, we need to ensure the comparison is case insensitive as well.
+// All columns in Snowflake are upper case by default, so we convert the expected columns to upper case as well.
+func (sf *snowflakeOfflineStore) checkSourceContainsResourceColumns(id ResourceID, schema ResourceSchema, opts ...ResourceOption) (bool, error) {
+	logger := sf.logger.WithResource(logging.ResourceType(id.Type.String()), id.Name, id.Variant).With("schema", schema)
```

Comment:
Feels like this logger could have been passed in? Also feels like context should be passed in even though its not being used currently